src/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -1025,5 +1025,46 @@ public static void MiscellaneousTest()
             Thread.SpinWait(1);
             Thread.Yield();
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void WindowsPrincipalPolicyTest_Windows()
+        {
+            DummyClass.RemoteInvoke(() =>
+            {
+                AppDomain.CurrentDomain.SetPrincipalPolicy(PrincipalPolicy.WindowsPrincipal);
+                Assert.Equal(Environment.UserDomainName + @"\" + Environment.UserName, Thread.CurrentPrincipal.Identity.Name);
```

Comment:
Is this test going to pass is the machine is not domain joined, or the current user is not a domain user? Perhaps skip for empty UsrrDomainName.